src/Runner.Worker/ExecutionContext.cs

```diff
@@ -273,23 +270,14 @@ public void RegisterPostJobStep(IStep step)
         /// add a child node, aka a step, to the current job to the Root.JobSteps based on the location. 
         /// </summary>
         public IStep CreateCompositeStep(
+            string scopeName,
             IActionRunner step,
             DictionaryContextData inputsData,
             Dictionary<string, string> envData)
         {
-            // If the context name is empty and the scope name is empty, we would generate a unique scope name for this child in the following format:
```

Comment:
@ethanchewy fyi i fixed a bug here. When ContextName was empty, each child scope name got a new guid (i.e. each nested step had a different scope name). I switched to pass the child scopeName as a parameter instead.

After the server generates default context names (M271-ish), we can eliminate the `scopeName` parameter since the outer step's context name will never be empty.

Reply:
ok gotcha - thanks!